packages/flutter/test/material/paginated_data_table_test.dart

```diff
@@ -102,6 +103,28 @@ void main() {
     expect(find.text('Eclair (0)'), findsNothing);
     expect(find.text('Gingerbread (0)'), findsNothing);
 
+    await tester.tap(find.byTooltip('Last page'));
+
+    expect(log, <String>['page-changed: 498']);
+    log.clear();
+
+    await tester.pump();
+
+    expect(find.text('Frozen yogurt (0)'), findsNothing);
+    expect(find.text('Donut (49)'), findsOneWidget);
+    expect(find.text('KitKat (49)'), findsOneWidget);
+
+    await tester.tap(find.byTooltip('First page'));
+
+    expect(log, <String>['page-changed: 0']);
+    log.clear();
+
+    await tester.pump();
+
+    expect(find.text('Frozen yogurt (0)'), findsOneWidget);
+    expect(find.text('Eclair (0)'), findsNothing);
+    expect(find.text('Gingerbread (0)'), findsNothing);
+
```

Comment:
I think a test to validate that buttons are disabled respectively, on the first and last pages, would be useful

Reply:
Makes sense! Adding it right now

Reply:
Done